### PR TITLE
Allow Unix Domain Socket addresses for ServiceEntry

### PR DESCRIPTION
**Pull request by spikecurtis**
Alternative to #468 

Signed-off-by: Spike Curtis <spike@tigera.io>

### CODE REVIEW

**Comment by saumoh** [comment]
specifying `unix` is fine. Once the need support Windows comes up we'll have to add their equivalent Ipc naming also.

**Comment by rshriram** [comment]
Do you have the corresponding pr in istio/istio changing all occurrences to use the new format?

**Comment by spikecurtis** [comment]
Not yet, but I will work on this immediately.

**Comment by spikecurtis** [comment]
@rshriram Ok, code is up on https://github.com/spikecurtis/istio/tree/service-entry-address

I'm not sure the "right" way to update the `vendor/istio.io/api` directory...

**Comment by rshriram** [comment]
Just change the gopkg.toml with your fork and do dep ensure -vendor-only

**Comment by rshriram** [comment]
Or start a PR to istio/istio

**Comment by spikecurtis** [comment]
@rshriram  https://github.com/istio/istio/pull/5479

File: authentication/v1alpha1/istio.authentication.v1alpha1.pb.html [hunk 1]
```diff
@@ -1,6 +1,5 @@
 ---
 title: Authentication Policy
-overview: Authentication policy for Istio services.
```
Comment (rshriram) [inline authentication/v1alpha1/istio.authentication.v1alpha1.pb.html]: ?.. where did this come from? Old Docker image for istio api ?

**Comment by rshriram** [comment]
Change should go to release 0.8

File: authentication/v1alpha1/istio.authentication.v1alpha1.pb.html (hunk 1 above)
Comment (spikecurtis) [reply authentication/v1alpha1/istio.authentication.v1alpha1.pb.html]: I'm using `gcr.io/istio-testing/protoc:2018-03-03` which is what is in the master Makefile...

**Comment by rshriram** [comment]
Use code from release 0.8 branch

**Comment by spikecurtis** [comment]
@rshriram fixed!

**Comment by spikecurtis** [comment]
@rshriram all required tests are now passing.

**Comment by spikecurtis** [comment]
We'll use `unix://` syntax and leave the proto as is. Will open a new docs-only PR.

**Pull request closed without merging**